packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -710,12 +723,20 @@ export class RestApi extends RestApiBase {
    * Import an existing RestApi that can be configured with additional Methods and Resources.
    */
   public static fromRestApiAttributes(scope: Construct, id: string, attrs: RestApiAttributes): IRestApi {
+    const restApiName = attrs.restApiName ?? '';
+
     class Import extends RestApiBase {
       public readonly restApiId = attrs.restApiId;
       public readonly restApiRootResourceId = attrs.rootResourceId;
       public readonly root: IResource = new RootResource(this, {}, this.restApiRootResourceId);
     }
 
+    class ImportWithApiName extends Import {
+      public readonly restApiName = restApiName;
+    }
+
+    if (attrs.restApiName !== undefined) return new ImportWithApiName(scope, id);
```

Comment:
we shouldn't be referencing `attrs.restApiName` here, we already have `restApiName` above. This usage is confusing; I would delete the `restApiName` local variable and use `attrs.restApiName` instead.